### PR TITLE
Delete cache item containing cache keys for a tag upon tag invalidation

### PR DESCRIPTION
**Pull request by Magomogo**
@eko what do you think, should we delete the collection of cache keys after each of the keys was deleted?

### CODE REVIEW

**Review by eko — COMMENTED** [review]
Hi @Magomogo,

I fully agree. Could you also please update the other stores in your pull request?

It appears that the tag key is only deleted on Redis stores.

Thank you!

**Comment by Magomogo** [comment]
Done for bigcache, risretto, memcache and pegasus. Freecache already has this functionality.

**Review by eko — APPROVED** [review]

**Comment by eko** [comment]
Thanks @Magomogo :)

File: store/go_cache.go [hunk 1]
```diff
@@ -132,11 +132,12 @@ func (s *GoCacheStore) Invalidate(ctx context.Context, options InvalidateOptions
 				cacheKeys = bytes
 			}
 
-			s.mu.RLock()
+			s.mu.Lock()
 			for cacheKey := range cacheKeys {
```
Comment (Magomogo) [inline store/go_cache.go:136]: Hello, @eko , I'm getting "concurrent map iteration and map write" error at this line in production. Have no clue why, could you please look, what I'm missing?

Comment (Magomogo) [reply store/go_cache.go:136]: Forget it, I had two instances of `GoCacheStore` by mistake.

Comment (eko) [reply store/go_cache.go:136]: Alright, this is cool that you could be able to find your issue :)

**Pull request merged**